tests/codex/testlogutils.nim

```diff
@@ -0,0 +1,214 @@
+    check loggedJson("t", "[\"formatted_a\",\"formatted_b\"]")
+
+  test "logs ref types":
+    let t = RefType(a: "a")
+    log t
+    check logged("t", "formatted_a")
+    check loggedJson("t", "\"a\"")
+
+  test "logs sequences of ref types":
+    let t1 = RefType(a: "a")
+    let t2 = RefType(a: "b")
+    let t = @[t1, t2]
+    log t
+    check logged("t", "\"@[formatted_a, formatted_b]\"")
+    check loggedJson("t", "[\"a\",\"b\"]")
+
+  test "logs distinct types":
+    let t = DistinctType(ObjectType(a: "a"))
+    log t
+    check logged("t", "formatted_a")
+    check loggedJson("t", "\"formatted_a\"")
+
+  test "logs sequences of distinct types":
+    let t1 = DistinctType(ObjectType(a: "a"))
+    let t2 = DistinctType(ObjectType(a: "b"))
+    let t = @[t1, t2]
+    log t
+    check logged("t", "\"@[formatted_a, formatted_b]\"")
+    check loggedJson("t", "[\"formatted_a\",\"formatted_b\"]")
+
+  test "formatIt can return non-string types":
+    let t = AnotherType(a: 1)
+    log t
+    check logged("t", "1")
+    check loggedJson("t", "1")
+
+  test "logs Option types":
+    let t = some ObjectType(a: "a")
+    log t
+    check logged("t", "some(formatted_a)")
+    check loggedJson("t", "\"formatted_a\"")
+
+  test "logs sequences of Option types":
+    let t1 = some ObjectType(a: "a")
+    let t2 = none ObjectType
+    let t = @[t1, t2]
+    log t
+    check logged("t", "\"@[some(formatted_a), none(ObjectType)]\"")
+    check loggedJson("t", """["formatted_a",null]""")
+
+  test "can define `$` override for T":
+    let o = ObjectType()
+    check $o == "used `$`"
+
+  test "logs NByte correctly":
+    let nb = 12345.NBytes
+    log nb
+    check logged("nb", "12345\'NByte")
+    check loggedJson("nb", "\"12345\'NByte\"")
+
+  test "logs BlockAddress correctly":
+    let cid = Cid.init("zb2rhgsDE16rLtbwTFeNKbdSobtKiWdjJPvKEuPgrQAfndjU1").tryGet
+    let ba = BlockAddress.init(cid, 0)
+    log ba
+    check logged("ba", "\"treeCid: zb2*fndjU1, index: 0\"")
+    check loggedJson("ba", """{"treeCid":"zb2rhgsDE16rLtbwTFeNKbdSobtKiWdjJPvKEuPgrQAfndjU1","index":0}""")
+
+  test "logs Cid correctly":
+    let cid = Cid.init("zb2rhmfWaXASbyi15iLqbz5yp3awnSyecpt9jcFnc2YA5TgiD").tryGet
+    log cid
+    check logged("cid", "zb2*A5TgiD")
+    check loggedJson("cid", "\"zb2rhmfWaXASbyi15iLqbz5yp3awnSyecpt9jcFnc2YA5TgiD\"")
+
+  test "logs StUint correctly":
+    let stint = 12345678901234.u256
+    log stint
+    check logged("stint", "12345678901234")
+    check loggedJson("stint", "\"12345678901234\"")
+
+  test "logs int correctly":
+    let int = 123
+    log int
+    check logged("int", "123")
+    check loggedJson("int", "123")
+
+  test "logs EthAddress correctly":
+    let address = EthAddress.fromHex("0xf75e076f650cd51dbfa0fd9c465d5037f22e1b1b")
+    log address
+    check logged("address", "0xf75e..1b1b")
+    check loggedJson("address", "\"0xf75e076f650cd51dbfa0fd9c465d5037f22e1b1b\"")
```

Comment:
@benbierens will changing these id formats for regular logs break any automations that you have? It sounds like many of these id's were shortened before but some weren't.

Reply:
> Member

This change (id formatting) won't break anything vital. Worst case it might fail a test, but I don't expect this and it would be an easy fix if it happened.